### PR TITLE
Added clearing of cache at restart.

### PR DESCRIPTION
**Pull request by crow-misia**
When using "Spring Boot Developer tools", since error occurs when restarting, cache is cleared after bean initialization.

### CODE REVIEW

**Comment by mslipper** [comment]
Hi @crow-misia,

Could you please provide some reproduction steps for this issue? Would like to understand it a bit more before reviewing.

**Comment by crow-misia** [comment]
Hi @mslipper

I will create a sample project, so please wait for a while.

**Comment by crow-misia** [comment]
Hi @mslipper 

I create reproduction code.(tried with Intellij IDEA)
https://github.com/crow-misia/jsonrpc4j-spring-restart-sample

steps
1. run from DemoApplication#main
2. api request with curl etc.
3. modify the code while running the application (For example, set "a + b" of CalcApiImpl to "a + b + 1")
4. run Build (For Intellij IDEA, execute "Build Project")
5. application restarts automatically.
6. again request api.
7. results
```json
{"jsonrpc":"2.0","id":"1","error":{"code":-32001,"message":"object is not an instance of declaring class","data":{"exceptionTypeName":"java.lang.IllegalArgumentException","message":"object is not an instance of declaring class"}}}
```

I hope it works.

**Comment by mslipper** [comment]
Thanks, @crow-misia. Will take a look this evening.

**Comment by mslipper** [comment]
Hi @crow-misia - I got tied up and haven't had a change to look at this yet. Rest assured that it's on my radar.

**Comment by crow-misia** [comment]
Hi @mslipper. It’s fine.thank you.

File: src/main/java/com/googlecode/jsonrpc4j/spring/AbstractJsonServiceExporter.java [hunk 1]
```diff
@@ -38,8 +38,9 @@
 	 * {@inheritDoc}
 	 */
 	public void afterPropertiesSet() throws Exception {
+		objectMapper = null;
 
-		if (objectMapper == null && applicationContext != null && applicationContext.containsBean("objectMapper")) {
+		if (applicationContext != null && applicationContext.containsBean("objectMapper")) {
```
Comment (mslipper) [inline src/main/java/com/googlecode/jsonrpc4j/spring/AbstractJsonServiceExporter.java]: What's the purpose of removing `objectMapper == null` from this if statement and instead setting it to `null` on line 41? What is the value of `objectMapper` when `afterPropertiesSet()` is called?

File: src/main/java/com/googlecode/jsonrpc4j/spring/JsonProxyFactoryBean.java [hunk 2]
```diff
@@ -54,38 +54,40 @@ public void afterPropertiesSet() {
...
-			if (objectMapper == null && applicationContext != null && applicationContext.containsBean("objectMapper")) {
-				objectMapper = (ObjectMapper) applicationContext.getBean("objectMapper");
-			}
-			if (objectMapper == null && applicationContext != null) {
-				try {
-					objectMapper = BeanFactoryUtils.beanOfTypeIncludingAncestors(applicationContext, ObjectMapper.class);
-				} catch (Exception e) {
-					logger.debug(e);
-				}
+		objectMapper = null;
```
Comment (mslipper) [inline src/main/java/com/googlecode/jsonrpc4j/spring/JsonProxyFactoryBean.java]: Same question here - what's the purpose of removing the null check and setting `objectMapper` to `null`?

File: src/main/java/com/googlecode/jsonrpc4j/spring/rest/JsonRestProxyFactoryBean.java [hunk 3]
```diff
@@ -56,40 +56,39 @@ public void afterPropertiesSet() {
...
-					objectMapper = BeanFactoryUtils.beanOfTypeIncludingAncestors(applicationContext, ObjectMapper.class);
-				} catch (Exception e) {
-					logger.debug(e);
-				}
-			}
-			
-			if (objectMapper == null) {
-				objectMapper = new ObjectMapper();
-			}
+		objectMapper = null;
```
Comment (mslipper) [inline src/main/java/com/googlecode/jsonrpc4j/spring/rest/JsonRestProxyFactoryBean.java]: Ditto here - `objectMapper` null check.

**Comment by mslipper** [comment]
Made a couple of comments. In general, I'm a little worried about what seem to be additional refactorings on top of the `ReflectionUtil.clearCache()` calls.

**Comment by crow-misia** [comment]
Hi @mslipper.

I thought that bean will be reused on restart.
But, I was wrong.

After checking again, when  afterPropertiesSet () is called, objectMapper was always null. 
Therefore, I reverted code and only code to clear the cache.

Thank you.

**Comment by mslipper** [comment]
LGTM. Thanks @crow-misia!

**Pull request merged**